### DISCUSSION TITLE
0.5 release discussion

### DISCUSSION THREAD

**Discussion by TimoGlastra** [openwallet-foundation/credo-ts#1525] [Architecture & Design]
A discussion to talk about the features we'd like to include in the 0.5.0 release. There's probably a lot of small breaking changes we'd like to bundle in this release.

## Breaking Changes

- #1510
- #1524 
- #1427
- #1638
- #1629

## Non-breaking changes
- #1550
- #1639
- did:peer:4 and DID Rotation support (https://github.com/hyperledger/aries-rfcs/pull/793)

## For 0.6

- #1539
- Extract out of band invitation to DidCommMessageRecord (with addition of OOB v2 and DidComm v2, breaking change)

**Comment by genaris** [comment]
I would really like to see both AnonCreds revocable credential issuance #1427 (no breaking change other than AnonCredsRegistry interfaces) and DIDComm V2 support. This last one is more challenging but I think I can spare some time starting next week to help @Artemkaaas on the needed changes and testing in order to speed-up the merging it.

**Comment by swcurran** [comment]
#1427 should be dependent on merging the replacement of Ursa with AnonCreds CL-Signatures — which is a PR that is out there — and getting that into AFJ. I recommend that be a priority.

**Comment by TimoGlastra** [comment]
I think we could get to a 0.5.0 release rather quickly. If we aim for revocation + didcomm v2 (+ some other breaking changes). I think that is enough new features already for a new release.

I think having the wallet api refactor would also be nice, but I can also live with it being pushed to 0.6

**Comment by TimoGlastra** [comment]
Seems like we're mostly there for a 0.5.0 release, there's a few todos left:
- Write migration documentation, I think we can get away with having no migration script for this release, which is really nice :)
- Do we want to include the live message pickup etc PR? I'm fine with pushing it to 0.6 and doing it quite soon
- we will do openid in 0.6 or 0.5.x
- We need to get stable releases for the shared components. I think indy-vdr and aries-askar are quite doable. For AnonCreds we have a bit of an issue with the new AnonCreds W3C work having been merged already. I'm thinking of creating a separate branch and doing a 0.2.x release of AnonCreds there. Then we can release 0.3.x with the new W3C functionality and have enough time to interate.

**Reply by genaris** [reply]
I agree that we'll need to get stable releases for shared components. And also if we didn't yet) remove the "experimental flag" we have for the related modules (otherwise the whole framework will be experimental, considering we don't have indy-sdk anymore).

I'll try to push some other changes in the pickup live PR in the incoming days: we've been testing it and works quite nice but there are some adjustments to do based on the feedback we got. I'd like to have it in 0.5 even if we consider it "experimental"... As long as it does not break current implementations (I guess most people are using "Implicit" or polling mode so it won't represent a major issue) besides the API changes 

Also would like to submit a PR I'm preparing for the DID rotation, to finish the step 1 of unqualified did migration stuff. But anyway it can be part of a 0.5.1.

**Comment by TimoGlastra** [comment]
From meeting notes:

- Credo 0.5.0 release
    Relevant features since v0.4.2:
    - Issue revocable AnonCreds credentials (https://github.com/openwallet-foundation/credo-ts/pull/1427)
    - did:peer:2 and did:peer:4 support in DID Exchange (https://github.com/openwallet-foundation/credo-ts/pull/1550)
    - DIF Presentation Exchange over DIDComm (Aries RFC 510) support (https://github.com/openwallet-foundation/credo-ts/pull/1676)
    - DID Rotation/Hang-up support (https://github.com/openwallet-foundation/credo-ts/pull/1699)
    - DID Caching (https://github.com/openwallet-foundation/credo-ts/pull/1710)
    - Message Pickup V2 Live Mode (https://github.com/openwallet-foundation/credo-ts/pull/1638)
    - OpenID4VC and SD-JWT-VC support (https://github.com/openwallet-foundation/credo-ts/pull/1708)
    - Removal of Indy SDK

  Remaining tasks:
    - Test 0.5.0 release in mobile wallet
    - Stable release for shared components
    - Announcement blog post
    - Announcing new version and rename on as much channels as possible
